lib/Sema/CSSimplify.cpp

```diff
@@ -9864,13 +9864,23 @@ bool ConstraintSystem::resolveClosure(TypeVariableType *typeVar,
         auto flags = param.getParameterFlags();
 
         // Note when a parameter is inferred to be isolated.
-        if (contextualParam->isIsolated() && !flags.isIsolated() && paramDecl)
+        if (contextualParam->isIsolated() && !flags.isIsolated() && paramDecl) {
           isolatedParams.insert(paramDecl);
 
+          // an 'isolated' param is also implicitly '_local',
+          // since in order to run "on" an actor, it must be a real local instance.
+          distributedKnownLocalParams.insert(paramDecl);
+        } else if (contextualParam->isDistributedKnownToBeLocal() &&
```

Comment:
I think this should be an `if` not an `else if`.

Reply:
Gotcha thanks!